File: src/main/java/io/github/saneea/dvh/feature/text/Text.java
package io.github.saneea.dvh.feature.text;

import java.util.Map;
import java.util.function.Supplier;

import io.github.saneea.dvh.Feature;
import io.github.saneea.dvh.FeatureContext;
import io.github.saneea.dvh.feature.multi.MultiFeatureBase;

public class Text extends MultiFeatureBase {

	@Override
	public Meta meta(FeatureContext context) {
		return Meta.from("plain text processing");
	}

	@Override
	public Map<String, Supplier<Feature>> getFeatureAliases() {
		return new AliasesBuilder()//
				.multiFeature(//
						"case", //
						"transform to upper/lower case", //
						new AliasesBuilder()//
								.feature("upper", ConvertTextCase.Upper::new)//
								.feature("lower", ConvertTextCase.Lower::new)//
								.build())//
				.multiFeature(//
						"chunker", //
						"split text to chunks or join chunks again", //
						new AliasesBuilder()//
								.feature("split", Split::new)//
								.feature("join", JoinLines::new)//
								.build())//
				.feature("toCharArray", ToCharArray::new)//
				.feature("trim", Trim::new)//
				.feature("newLine", AddNewLine::new)//
				.feature("pipe", Pipe::new)//
				.feature("slash", SlashReplacer::new)//
				.build();
	}
}

File: src/main/java/io/github/saneea/dvh/feature/text/Text.kt
package io.github.saneea.dvh.feature.text

import io.github.saneea.dvh.Feature.Meta
import io.github.saneea.dvh.FeatureContext
import io.github.saneea.dvh.feature.multi.FeaturesCreators
import io.github.saneea.dvh.feature.multi.MultiFeatureBase

class Text : MultiFeatureBase() {

    override val meta get() = Meta("plain text processing")

    override fun getFeatureAliases(): FeaturesCreators =
        AliasesBuilder()
            .multiFeature(
                "case",
                "transform to upper/lower case",
                AliasesBuilder()
                    .feature("up", ConvertTextCase::Upper)
                    .feature("down", ConvertTextCase::Lower)
                    .build()
            )
            .multiFeature(
                "chunker",
                "split text to chunks or join chunks again",
                AliasesBuilder()
                    .feature("split", ::Split)
                    .feature("join", ::JoinLines)
                    .build()
            )
            .feature("toCharArray", ::ToCharArray)
            .feature("trim", ::Trim)
            .feature("newLine", ::AddNewLine)
            .feature("pipe", ::Pipe)
            .feature("slash", ::SlashReplacer)
            .build()
}